src/litdata/utilities/dataset_utilities.py

```diff
@@ -91,8 +92,41 @@ def _should_replace_path(path: Optional[str]) -> bool:
     return path.startswith("/teamspace/datasets/") or path.startswith("/teamspace/s3_connections/")
 
 
+def _read_last_updated_timestamp(input_dir: Optional[Dir]) -> str:
+    """Read last updated timestamp from index.json file."""
+    last_updation_timestamp = ""
+    index_json_content = None
+    assert isinstance(input_dir, Dir)
+
+    if input_dir.path is not None and os.path.exists(os.path.join(input_dir.path, _INDEX_FILENAME)):
+        # read index.json file and read last_updation_timestamp
+        index_json_content = load_index_file(input_dir.path)
+    elif input_dir.url is not None:
+        assert input_dir.url is not None
+        # download index.json file and read last_updation_timestamp
+        with tempfile.TemporaryDirectory() as tmp_directory:
+            temp_index_filepath = os.path.join(tmp_directory, _INDEX_FILENAME)
+            downloader = get_downloader_cls(input_dir.url, input_dir.path, [])  # type: ignore
```

Comment:
Could we use `tmp_directory` as the cache directory here instead of `input_dir.path`?

> Reason: The downloader might receive None as the cache directory. If the downloader uses this cache directory and finds it empty, it defaults to using the standard downloader cache, which could lead to a FileNotFoundError.

Reply:
I also noticed, in the first run, the index file get's downloaded twice. 
<img width="1065" alt="image" src="https://github.com/user-attachments/assets/41788e75-053f-4f1a-8e91-661e0b84dc8c">

To reproduce maybe, we can test with s3 uploaded dataset. In my case, I was using the huggingface.(this feature is not available yet.)